common/trinary/flex_trit.c

```diff
@@ -166,12 +166,42 @@ size_t flex_trits_to_trytes(tryte_t *trytes, size_t to_len,
 #if defined(FLEX_TRIT_ENCODING_1_TRIT_PER_BYTE)
   trits_to_trytes((trit_t *)flex_trits, trytes, num_trits);
 #elif defined(FLEX_TRIT_ENCODING_3_TRITS_PER_BYTE)
-  memcpy(trytes, flex_trits, num_flex_trits_for_trits(num_trits));
+  uint8_t residual = num_trits % 3U;
+  size_t num_bytes = num_flex_trits_for_trits(num_trits);
+  // There is a risk of noise after the last trit so we need to clean up
+  if (residual) {
+    trit_t trits[3];
+    num_bytes--;
+    trytes_to_trits((tryte_t *)(flex_trits + num_bytes), trits, 1);
+    trits_to_trytes(trits, (tryte_t *)(trytes + num_bytes), residual);
+  }
+  memcpy(trytes, flex_trits, num_bytes);
 #elif defined(FLEX_TRIT_ENCODING_4_TRITS_PER_BYTE) || \
-    defined(FLEX_TRIT_ENCODING_5_TRITS_PER_BYTE)
-  trit_t trits[num_trits];
-  flex_trits_to_trits(trits, num_trits, flex_trits, len, num_trits);
-  trits_to_trytes((trit_t *)trits, trytes, num_trits);
+      defined(FLEX_TRIT_ENCODING_5_TRITS_PER_BYTE)
+  union _shifter {
+    uint64_t val;
+    trit_t trits[8];
+  };
+  union _shifter shifter = { 0 };
+  size_t offset = 0;
+  size_t max_trits, trits_for_tryte;
+  for (int i = 0, j = 0; num_trits || offset; j++) {
+    if (offset < 3) {
+      max_trits = MIN(NUM_TRITS_PER_FLEX_TRIT, num_trits);
+      flex_trits_to_trits(shifter.trits + offset, max_trits, &flex_trits[i++],
+                          max_trits, max_trits);
+      num_trits -= max_trits;
+      offset += max_trits;
+    }
+    trits_for_tryte = MIN(3, offset);
+    trits_to_trytes(shifter.trits, &trytes[j], trits_for_tryte);
+#if BYTE_ORDER == LITTLE_ENDIAN
+    shifter.val = shifter.val >> 24;
```

Comment:
where does `shifter.val` being used?

Reply:
It is used to shift the bytes of `shifter.trits`